mne/preprocessing/stim.py

```diff
@@ -84,8 +84,7 @@ def fix_stim_artifact(inst, events=None, event_id=None, tmin=0.,
     window = None
     if mode == 'window':
         window = _get_window(s_start, s_end)
-    ch_names = inst.info['ch_names']
-    picks = pick_channels(ch_names, ch_names)
+    picks = _pick_data_channels(inst.info)
```

Comment:
I forgot to mention this, but I also fixed this. Previously `fix_stim_artifact` was fixing all channels (this line was equivaletn to `picks = np.arange(len(ch_names))`), but now it should only fix the data channels. `whats_new.rst` updated.